File: submissions/python3/532.py
class Solution:
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        counts = {}
        for n in nums:
            if n not in counts:
                counts[n] = 1
            else:
                counts[n] += 1
        ret = 0
        for c in counts:
            if k == 0 and counts[c] > 1:
                ret += 1
            elif k > 0 and c - k in counts:
                ret += 1
        return ret

File: submissions/cpp/532.cpp
class Solution {
public:
    int findPairs(vector<int> &nums, int k) {
        map<int, int> knums;
        int ret = 0;
        for (auto num: nums)
            ++knums[num];
        for (auto m:knums) {
            if (k == 0 && m.second > 1)
                ++ret;
            else if (k > 0 && knums.find(m.first - k) != knums.end())
                ++ret;
        }
        return ret;
    }
};
